src/averageAges.js

```diff
@@ -1,61 +1,47 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function getAverageAge(ages) {
+  return ages
+    .map(user => user.died - user.born)
+    .reduce((acc, user) => acc + user) / ages.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mans = people.filter(user => user.sex === 'm');
+
+  if (century !== undefined) {
+    mans = mans.filter(man => Math.ceil(man.died / 100) === century);
+  }
+
+  return getAverageAge(mans);
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(user => user.sex === 'f');
+
+  if (withChildren) {
+    womens = people
+      .filter(user => people.some(little => little.mother === user.name)
+      );
+  }
```

Comment:
This looks good but if you want to go beyond, you can check both conditions in filter method using some logical operators and ternary operators.